Support disconnected network environments

dprince opened the pull request:
This PR adds support for installing the operator in disconnected network environments. To build with image-digests set USE_IMAGE_DIGESTS=true before running make bundle.

For Prow jobs we are enabling this via .prow-ci.env

This drops the old logic from create_bundle.sh which has been broken with operator-sdk's make bundle for some time.

(NOTE: this currently requires a secure registry)

gibizer commented on api/v1beta1/placementapi_types.go:
```diff
@@ -107,15 +108,13 @@ type PlacementAPISpec struct {
 type PasswordSelector struct {
-	// +kubebuilder:validation:Optional
-	// +kubebuilder:default="PlacementDatabasePassword"
+	// +kubebuilder:validation:Required
```
This prevents the client to only specify `service` but let `database` to be defaulted.

I think we can keep these Optional.

If we add the L72 that actually solves the case when the whole `passwordSelectors` fields is missing from the input.

stuggi replied:
right if we want to be able to set each selector individual we should keep those as optional and default in the struct as well. Will update

stuggi replied:
done

dprince committed:
Support disconnected network environments

This PR adds support for installing the operator in disconnected
network environments. To build with image-digests set
USE_IMAGE_DIGESTS=true before running make bundle.

For Prow jobs we are enabling this via .prow-ci.env

This drops the old logic from create_bundle.sh which has been
broken with operator-sdk's make bundle for some time.

(NOTE: this currently requires a secure registry)

gibizer reviewed (approved):
looks good to me
